core/types/state_account.go

```diff
@@ -48,6 +48,6 @@ type isMultiCoin bool
 
 var IsMultiCoinPayloads = ethtypes.RegisterExtras[isMultiCoin]()
 
-func IsMultiCoin(a *StateAccount) bool {
-	return bool(IsMultiCoinPayloads.FromStateAccount(a))
+func IsMultiCoin(a *SlimAccount) bool {
```

Comment:
(optional) You could use `types.ExtraPayloadCarrier` as the parameter type to accept either.